lib/redis.js

```diff
@@ -64,7 +64,12 @@ function isAbortError(err) {
 function isClient(object) {
   if (!object || typeof object !== 'object') return false;
   const name = object.constructor.name;
-  return name === 'Redis' || name === 'RedisClient' || name === 'Cluster';
+  return (
+    name === 'Redis' ||
+    name === 'RedisClient' ||
+    name === 'Cluster' ||
+    name === 'RedisMock'
+  );
```

Comment:
Would it be cleaner to refactor to collection based check - i.e.: 
```
[
    'Redis',
    'RedisClient',
    'Cluster',
    'RedisMock',
].includes(name)
```

This opens up a possibility to accept the extension of the implicit list as on of the options, allowing for generic solution where support for different Redis client implementations is extended on the client side.

Alternatively client could proved explicit option flag indicating that provided `client` is a client rather than settings. For backwards compatibility would still have to support list of known class names.